avaje-config/src/main/java/io/avaje/config/Parsers.java

```diff
@@ -23,13 +23,21 @@ final class Parsers {
   }
 
   private void initYamlParser() {
-    YamlLoader yamlLoader;
-    try {
-      Class.forName("org.yaml.snakeyaml.Yaml");
-      yamlLoader = new YamlLoaderSnake();
-    } catch (ClassNotFoundException e) {
-      yamlLoader = new YamlLoaderSimple();
-    }
+    var modules = ModuleLayer.boot();
+    YamlLoader yamlLoader =
+        modules
+            .findModule("io.avaje.config")
+            .map(m -> modules.findModule("org.yaml.snakeyaml").isPresent())
```

Comment:
Why bother with finding the module?

Reply:
If you mean why we are checking if the current module exists, the reason is so that we can tell whether we are running on the module path. (It returns empty when running on the class path)